src/Graphics/ColorBufferReader.cpp

```diff
@@ -0,0 +1,89 @@
+
+#include "ColorBufferReader.h"
+#include "FramebufferTextureFormats.h"
+#include "Context.h"
+#include "Parameters.h"
+#include <algorithm>
+#include <cstring>
+
+namespace graphics {
+
+	ColorBufferReader::ColorBufferReader(CachedTexture * _pTexture)
+		: m_pTexture(_pTexture)
+	{
+		m_pixelData.resize(m_pTexture->textureBytes);
+		m_tempPixelData.resize(m_pTexture->textureBytes);
+	}
+
+	const u8* ColorBufferReader::_convertFloatTextureBuffer(const u8* _gpuData, u32 _width, u32 _height,
+		u32 _heightOffset, u32 _stride)
+	{
+		std::copy_n(_gpuData, m_tempPixelData.size(), m_tempPixelData.data());
+		u8* pixelDataAlloc = m_pixelData.data();
+		float* pixelData = reinterpret_cast<float*>(m_tempPixelData.data());
+		const int colorsPerPixel = 4;
```

Comment:
why you use signed integer for obviously unsigned values?  colorsPerPixel , widthPixels, stridePixels can't be negative